model/src/main/scala/aqua/model/ScriptModel.scala

```diff
@@ -1,24 +1,36 @@
 package aqua.model
 
 import aqua.model.func.{FuncCallable, FuncModel}
-import cats.Monoid
+import cats.{Monoid, Semigroup}
 import cats.data.Chain
 
 case class ScriptModel(
   funcs: Chain[FuncModel] = Chain.empty,
   services: Chain[ServiceModel] = Chain.empty,
-  types: Chain[TypeModel] = Chain.empty
+  types: Chain[TypeModel] = Chain.empty,
+  constants: Chain[ConstantModel] = Chain.empty
 ) extends Model {
 
+  case class Acc(
+    arrows: Map[String, FuncCallable],
+    values: Map[String, ValueModel]
+  )
+
   def resolveFunctions: Chain[FuncCallable] =
     funcs
-      .foldLeft((Map.empty[String, FuncCallable], Chain.empty[FuncCallable])) {
-        case ((funcsAcc, outputAcc), func) =>
-          val fr = func.captureArrows(funcsAcc)
-          funcsAcc.updated(func.name, fr) -> outputAcc.append(fr)
+      .foldLeft(
+        (
+          Acc(arrows = Map.empty, values = constants.map(c => c.name -> c.value).toList.toMap),
+          Chain.empty[FuncCallable]
+        )
+      ) { case ((acc, outputAcc), func) =>
+        val fr = func.captureParts(acc.arrows, acc.values)
+        Acc(
+          acc.arrows.updated(func.name, fr),
+          acc.values ++ fr.capturedValues
```

Comment:
This `++` acts as an export from the function. What can be exported from it? Constants can't be defined inside functions. Return values should not be passed anywhere, because it's a definition of the function, not its execution.